packages/devtools_app/lib/src/screens/memory/panes/diff/widgets/snapshot_list.dart

```diff
@@ -128,16 +128,19 @@ class _SnapshotListItemsState extends State<_SnapshotListItems>
     super.didChangeDependencies();
 
     _currentSnapshotsLength = widget.controller.snapshots.value.length;
-    addAutoDisposeListener(widget.controller.snapshots, () {
+
+    addAutoDisposeListener(widget.controller.snapshots);
+
+    addAutoDisposeListener(widget.controller.selectedIndex, () async {
+      setState(() {});
+
       final newSnapshotsLength = widget.controller.snapshots.value.length;
       if (newSnapshotsLength > _currentSnapshotsLength) {
-        // If new snapshot added, scroll to bottom.
-        setState(() => _scrollController.autoScrollToBottom());
+        _currentSnapshotsLength = newSnapshotsLength;
+        await _scrollController.autoScrollToBottom();
+        setState(() {});
```

Comment:
couple things here:
1) avoid empty setState calls if you can. Whatever values need to be updated or actions that need to be taken should take place inside the setState callback. 
2) we are potentially calling setState twice here. Instead, place all this code inside a single setState callback. 
```
setState(() {
  final newSnapshotsLength = widget.controller.snapshots.value.length;
  if (newSnapshotsLength > _currentSnapshotsLength) {
    _currentSnapshotsLength = newSnapshotsLength;
    await _scrollController.autoScrollToBottom();
  }
});
```